### ISSUE TITLE
False positives by UnusedVariable on unnamed variables (Java 22)

### ISSUE THREAD

**Issue by ksobolew** [google/error-prone#4451]
Compiling code like this:

```java
var _ = someExpression();
```

Results in this report from Error Prone (2.28.0):

```
java: [UnusedVariable] The local variable '' is never read.
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean to remove this line?
```

This is a new feature in Java 22, which can be used to deliberately ignore a return value e.g. from methods declared as "must use return value". Note that the `UnusedVariable` check works fine in this case if the variable is called "ignored" or something like that.
